### PR TITLE
P42-217: Connect the Codemagic CI to build the main project automatically

### PR DESCRIPTION
**Pull request by siretjorro**

### CODE REVIEW

File: codemagic.yaml [hunk 1]
```diff
@@ -0,0 +1,222 @@
+# Has a separate flows for develop branch (automatic builds),
+# Merge Request (automatic builds) and master branch Release builds (manual builds).
```
Comment (harri35) [inline codemagic.yaml:2]: The "and master branch Release builds (manual builds)." I think you can release as e do not release any artifacts from here

File: codemagic.yaml [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+      events:
+        - pull_request
+      branch_patterns:
+        - pattern: 'develop'
+          include: true
+          source: false
+      cancel_previous_builds: true
+    scripts:
+      - *stepLibLocalSetup
+      - *stepLibCheckCode
```
Comment (harri35) [inline codemagic.yaml:216]: I understand that the lib only has sample tests? Jus that if "develop-lib-builds" runs tests also then it would probably be safer to also run these for PRs. Or if we do not care about the tests because they are just samples then maybe not run them at either.

**Review by harri35 — APPROVED** [review]
Looks fine to me. It can be merged when comments are addressed
Nice work!

**Comment by harri35** [comment]
Ah, didn't notice before - JIRA also has this criteria:
> Update Developing and updating the template itself section in README to include a step to verify that builds succeed before creating a merge request
Or maybe rephrase it a little? Because CM autobuilds for a PR and then anyone (even non-employee) will see if the build succeeds or not. Before the PR only Lab employee will see it

File: codemagic.yaml (hunk 2 above)
Comment (siretjorro) [reply codemagic.yaml:216]: Added tests to PR workflow as well.
